spring-rabbit/src/main/java/org/springframework/amqp/rabbit/config/AbstractRabbitListenerContainerFactory.java

```diff
@@ -356,30 +383,33 @@ public C createListenerContainer(RabbitListenerEndpoint endpoint) {
-				.acceptIfNotNull(this.transactionManager, instance::setTransactionManager)
-				.acceptIfNotNull(this.prefetchCount, instance::setPrefetchCount)
-				.acceptIfNotNull(this.defaultRequeueRejected, instance::setDefaultRequeueRejected)
-				.acceptIfNotNull(this.adviceChain, instance::setAdviceChain)
-				.acceptIfNotNull(this.recoveryBackOff, instance::setRecoveryBackOff)
-				.acceptIfNotNull(this.mismatchedQueuesFatal, instance::setMismatchedQueuesFatal)
-				.acceptIfNotNull(this.missingQueuesFatal, instance::setMissingQueuesFatal)
-				.acceptIfNotNull(this.consumerTagStrategy, instance::setConsumerTagStrategy)
-				.acceptIfNotNull(this.idleEventInterval, instance::setIdleEventInterval)
-				.acceptIfNotNull(this.failedDeclarationRetryInterval, instance::setFailedDeclarationRetryInterval)
-				.acceptIfNotNull(this.applicationEventPublisher, instance::setApplicationEventPublisher)
-				.acceptIfNotNull(this.autoStartup, instance::setAutoStartup)
-				.acceptIfNotNull(this.phase, instance::setPhase)
-				.acceptIfNotNull(this.afterReceivePostProcessors, instance::setAfterReceivePostProcessors);
+			.acceptIfNotNull(this.acknowledgeMode, instance::setAcknowledgeMode)
+			.acceptIfNotNull(this.channelTransacted, instance::setChannelTransacted)
+			.acceptIfNotNull(this.applicationContext, instance::setApplicationContext)
+			.acceptIfNotNull(this.taskExecutor, instance::setTaskExecutor)
+			.acceptIfNotNull(this.transactionManager, instance::setTransactionManager)
+			.acceptIfNotNull(this.prefetchCount, instance::setPrefetchCount)
+			.acceptIfNotNull(this.defaultRequeueRejected, instance::setDefaultRequeueRejected)
+			.acceptIfNotNull(this.adviceChain, instance::setAdviceChain)
+			.acceptIfNotNull(this.recoveryBackOff, instance::setRecoveryBackOff)
+			.acceptIfNotNull(this.mismatchedQueuesFatal, instance::setMismatchedQueuesFatal)
+			.acceptIfNotNull(this.missingQueuesFatal, instance::setMissingQueuesFatal)
+			.acceptIfNotNull(this.consumerTagStrategy, instance::setConsumerTagStrategy)
+			.acceptIfNotNull(this.idleEventInterval, instance::setIdleEventInterval)
+			.acceptIfNotNull(this.failedDeclarationRetryInterval, instance::setFailedDeclarationRetryInterval)
+			.acceptIfNotNull(this.applicationEventPublisher, instance::setApplicationEventPublisher)
+			.acceptIfNotNull(this.autoStartup, instance::setAutoStartup)
+			.acceptIfNotNull(this.phase, instance::setPhase)
+			.acceptIfNotNull(this.afterReceivePostProcessors, instance::setAfterReceivePostProcessors);
+		instance.setDeBatchingEnabled(!this.batchListener);
 		if (endpoint != null) { // endpoint settings overriding default factory settings
 			javaUtils
-					.acceptIfNotNull(endpoint.getAutoStartup(), instance::setAutoStartup)
-					.acceptIfNotNull(endpoint.getTaskExecutor(), instance::setTaskExecutor);
+				.acceptIfNotNull(endpoint.getAutoStartup(), instance::setAutoStartup)
+				.acceptIfNotNull(endpoint.getTaskExecutor(), instance::setTaskExecutor);
+			javaUtils
```

Comment:
Looks like this can be combined with the previous chain.

Reply:
I thought it was a bit dirty to do that since the property receiver is a different object; I kept changing my mind but ended up with this.

Feel free to change if you feel strongly.